com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -107,6 +107,44 @@ public bool UseLocal
             set => m_UseLocal.Value = value;
         }
 
+        /// <summary>
+        /// Updates the NetworkTransform's authority model at runtime.
+        /// </summary>
+        /// <param name="newAuthority"></param>
+        public void SetAuthority(Authority newAuthority)
+        {
+            TransformAuthority = newAuthority;
+            UpdateVarPermissions();
+            // todo this should be synced with the other side. let's wait for a more final solution before adding more code here
+        }
+
+        private void UpdateVarPermissions()
+        {
+            switch (TransformAuthority)
+            {
+                case Authority.Client:
+                    m_NetworkPosition.Settings.WritePermission = NetworkVariablePermission.OwnerOnly;
```

Comment:
Almost feels like we might want a wrapper for these var updates in case more are added later.